homeassistant/components/volumio/browse_media.py

```diff
@@ -0,0 +1,157 @@
+"""Support for media browsing."""
+import json
+
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_ALBUM,
+    MEDIA_CLASS_ARTIST,
+    MEDIA_CLASS_CHANNEL,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_GENRE,
+    MEDIA_CLASS_PLAYLIST,
+    MEDIA_CLASS_TRACK,
+    MEDIA_TYPE_MUSIC,
+)
+
+PLAYABLE_ITEM_TYPES = [
+    "folder",
+    "song",
+    "mywebradio",
+    "webradio",
+    "playlist",
+    "cuesong",
+    "remdisk",
+    "cuefile",
+    "folder-with-favourites",
+    "internal-folder",
+]
+
+NON_EXPANDABLE_ITEM_TYPES = [
+    "song",
+    "webradio",
+    "mywebradio",
+    "cuesong",
+    "album",
+    "artist",
+    "cd",
+    "play-playlist",
+]
+
+
+def _item_to_children_media_class(item, info=None):
+    if info and "album" in info and "artist" in info:
+        return MEDIA_CLASS_TRACK
+    if item["uri"].startswith("playlists"):
+        return MEDIA_CLASS_PLAYLIST
+    if item["uri"].startswith("artists://"):
+        if len(item["uri"]) > 10:
+            return MEDIA_CLASS_ALBUM
+        return MEDIA_CLASS_ARTIST
+    if item["uri"].startswith("albums://"):
+        if len(item["uri"]) > 9:
+            return MEDIA_CLASS_TRACK
+        return MEDIA_CLASS_ALBUM
+    if item["uri"].startswith("genres://"):
+        if len(item["uri"]) > 9:
```

Comment:
How can you use a length to check if it's an album. Can't you do a startswith check ?

Reply:
if it's `genres://` it's a list of genres. When you navigate further, it's a list of albums Since there isn't a pre-determined list of genres, all you know is that it starts with `genres://` but is longer (hence `len >9`). When you navigate into the album, you'll hit the first condition (`if info and "album" in info and "artist" in info:`)

Reply:
That's exactly the condition in line 54 :)

Reply:
@balloob I think I understand the source of confusion. The code you're referring to is already nested inside a `item["uri"].startswith("genres://")` condition, which you might not have noticed.

Reply:
The logic might become more clear if we use constants for the string we use in the starts with check and use the same constant in the length check.